demos/logging-stack/logging_stack.h

```diff
@@ -72,57 +72,118 @@
+            #define LogError( message )    SdkLog( ( "[ERROR] "LOG_METADATA_FORMAT, LOG_METADATA_ARGS ) ); SdkLog( message ); SdkLog( ( "\r\n" ) )
+            #define LogWarn( message )     SdkLog( ( "[WARN] "LOG_METADATA_FORMAT, LOG_METADATA_ARGS ) ); SdkLog( message ); SdkLog( ( "\r\n" ) )
+            #define LogInfo( message )
+            #define LogDebug( message )
+
+        #elif LIBRARY_LOG_LEVEL == LOG_ERROR
+            /* Only ERROR messages will be logged. */
+            #define LogError( message )    SdkLog( ( "[ERROR] "LOG_METADATA_FORMAT, LOG_METADATA_ARGS ) ); SdkLog( message ); SdkLog( ( "\r\n" ) )
+            #define LogWarn( message )
+            #define LogInfo( message )
+            #define LogDebug( message )
+
+        #else /* if LIBRARY_LOG_LEVEL == LOG_ERROR */
+
+            #define LogError( message )
+            #define LogWarn( message )
+            #define LogInfo( message )
+            #define LogDebug( message )
+
+        #endif /* if LIBRARY_LOG_LEVEL == LOG_ERROR */
+    #endif /* if !defined( LIBRARY_LOG_LEVEL ) || ( ( LIBRARY_LOG_LEVEL != LOG_NONE ) && ( LIBRARY_LOG_LEVEL != LOG_ERROR ) && ( LIBRARY_LOG_LEVEL != LOG_WARN ) && ( LIBRARY_LOG_LEVEL != LOG_INFO ) && ( LIBRARY_LOG_LEVEL != LOG_DEBUG ) ) */
+#endif /* ifndef DOXYGEN */
+
+
+/* The macro definition for logging interface is for Doxygen
+ * documentation only. */
+
+/**
+ * @brief Definition of logging interface macro that logs messages at the "Debug"
+ * level, when debug level logging is enabled.
+ *
+ * This macro is only enabled for #LOG_DEBUG level configuration in this
+ * logging stack implementation.
+ */
+#ifdef DOXYGEN
+    #define LogDebug( message )    SdkLog( ( "[DEBUG] "LOG_METADATA_FORMAT, LOG_METADATA_ARGS ) ); SdkLog( message ); SdkLog( ( "\r\n" ) )
+#endif
+
+/**
+ * @brief Definition of logging interface macro that logs messages at the "Info"
+ * level, when info level logging is enabled.
+ *
+ * This macro is only enabled for #LOG_DEBUG and #LOG_INFO level configurations
+ * in this logging stack implementation.
+ */
+#ifdef DOXYGEN
+    #define LogInfo( message )    SdkLog( ( "[INFO] "LOG_METADATA_FORMAT, LOG_METADATA_ARGS ) ); SdkLog( message ); SdkLog( ( "\r\n" ) )
+#endif
+
+/**
+ * @brief Definition of logging interface macro that logs messages at the "Warning"
+ * level, when warning level logging is enabled.
+ *
+ * This macro is only enabled for #LOG_DEBUG, #LOG_INFO and #LOG_WARN level
+ * configurations in this logging stack implementation.
+ */
+#ifdef DOXYGEN
+    #define LogWarn( message )    SdkLog( ( "[WARN] "LOG_METADATA_FORMAT, LOG_METADATA_ARGS ) ); SdkLog( message ); SdkLog( ( "\r\n" ) )
+#endif
+
+/**
+ * @brief Definition of logging interface macro that logs messages at the "Error"
+ * level, when error level logging is enabled.
+ *
+ * This macro is only enabled for all logging level configurations
+ * unless except the #LOG_NONE configuration.
+ */
+#ifdef DOXYGEN
```

Comment:
If we're going to define these specifically for doxygen, I'd just put them all in one block instead of repeating this `#ifdef`

Reply:
Enclosed all macros within a single `#ifdef`